pkg/machines/components/nicAdd.jsx

```diff
@@ -0,0 +1,242 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import cockpit from 'cockpit';
+import PropTypes from 'prop-types';
+import {
+    Modal,
+    Button
+} from 'patternfly-react';
+
+import { ModalError } from 'cockpit-components-inline-notification.jsx';
+import { getNetworkDevices } from '../helpers.js';
+import { NICBody } from './nicBody.jsx';
+import {
+    getVm
+} from '../actions/provider-actions.js';
+import { attachIface } from '../libvirt-dbus.js';
+
+import './nicEdit.css';
+import 'form-layout.less';
+
+const _ = cockpit.gettext;
+
+const NetworkMacRow = ({ idPrefix, dialogValues, onValueChanged }) => {
+    return (
+        <>
+            <label className='checkbox-inline'>
+                <input id={`${idPrefix}-set-mac`}
+                    type='checkbox'
+                    checked={dialogValues.setNetworkMac}
+                    onChange={e => onValueChanged('setNetworkMac', !dialogValues.setNetworkMac)} />
+                {_("Set MAC address")}
```

Comment:
Is this a child of `Model`? The label makes it look like it does.
Maybe better to put it next to the `Mac Address` label to make the grouping more clear.
Also, what happens if I do not chck this? It doesn't get a mac address?

Reply:
No MAC address is not child of Model and not related to it.

MAC address is optional input. So if you choose not to set it, libvirt will autogenerate MAC address for you.
But there are use cases for allowing user to set MAC address (e.g. he might have DNS configuration with IPs statically assigned to some pre-existing MACs)

Reply:
You mean putting it next to `MAC Address` label similar to how `Custom mount options` is next to input fields in **New NFS Mount** in Storage page?

Reply:
Yes, like this:
![64266012-2c4cf200-cf34-11e9-9125-c346569ff3a5](https://user-images.githubusercontent.com/209895/64324450-b3e34100-cfc6-11e9-8942-68836d973ecb.png)

Reply:
Done. Repushed.